src/breadcrumbs/index.ts

```diff
@@ -24,6 +24,14 @@ class Breadcrumbs {
       labels: this.props.labels,
       showLabels: this.props.showLabels
     };
+
+    const originalQuery = this.select(Selectors.query);
+    const correctedQuery = this.select(Selectors.currentQuery);
+    if (originalQuery !== correctedQuery) {
```

Comment:
Is there a way to refactor this so it only causes 1 update? Rather than 1 for corrected query and 2nd one for original query